compiler/rustc_parse/src/parser/diagnostics.rs

```diff
@@ -298,7 +298,7 @@ struct BadQPathStage2 {
 #[diag(parser::incorrect_semicolon)]
 struct IncorrectSemicolon<'a> {
     #[primary_span]
-    #[suggestion_short(applicability = "machine-applicable")]
+    #[suggestion_short(code = "", applicability = "machine-applicable")]
```

Comment:
Should we make the "suggestion type" an argument, like `#[suggestion(code = "", applicability = "machine-applicable", short)`?

Also, why is `code =` now mandatory? (I'm looking at the code to try and answer that question for myself, but failing.)

Reply:
> why is `code =` now mandatory? 

I introduced that in #101851, both for consistency with `Subdiagnostic` (which has always required it) and for clarity (what does a suggestion without any replacement code mean?)

Reply:
Makes sense!